api/src/main/java/org/pmiops/workbench/db/model/DbInstitution.java

```diff
@@ -11,88 +12,128 @@
+    this.institutionId = institutionId;
+    return this;
+  }
+
+  @Column(name = "short_name", nullable = false, unique = true)
   public String getShortName() {
     return shortName;
   }
 
-  public void setShortName(String shortName) {
+  public DbInstitution setShortName(String shortName) {
     this.shortName = shortName;
+    return this;
   }
 
+  @Column(name = "display_name", nullable = false)
   public String getDisplayName() {
     return displayName;
   }
 
-  public void setDisplayName(String displayName) {
+  public DbInstitution setDisplayName(String displayName) {
     this.displayName = displayName;
+    return this;
   }
 
-  public Short getOrganizationTypeEnum() {
-    return organizationTypeEnum;
+  @Column(name = "organization_type_enum")
+  public OrganizationType getOrganizationTypeEnum() {
+    return DbStorageEnums.organizationTypeFromStorage(organizationTypeEnum);
   }
 
-  public void setOrganizationTypeEnum(Short organizationTypeEnum) {
-    this.organizationTypeEnum = organizationTypeEnum;
+  public DbInstitution setOrganizationTypeEnum(OrganizationType type) {
+    this.organizationTypeEnum = DbStorageEnums.organizationTypeToStorage(type);
+    return this;
   }
 
+  @Column(name = "organization_type_other_text")
   public String getOrganizationTypeOtherText() {
     return organizationTypeOtherText;
   }
 
-  public void setOrganizationTypeOtherText(String organizationTypeOtherText) {
+  public DbInstitution setOrganizationTypeOtherText(String organizationTypeOtherText) {
     this.organizationTypeOtherText = organizationTypeOtherText;
+    return this;
   }
 
+  @OneToMany(mappedBy = "institution", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+  @NotNull
   public Set<DbInstitutionEmailDomain> getEmailDomains() {
     return emailDomains;
   }
 
-  public void setEmailDomains(Iterable<DbInstitutionEmailDomain> emailDomains) {
-    this.emailDomains =
-        StreamSupport.stream(emailDomains.spliterator(), false).collect(Collectors.toSet());
+  /**
+   * Effectively: do an in-place this.emailDomains = emailDomains
+   *
+   * <p>Hibernate doesn't like it when you reassign collections. Instead, modify in-place. First,
+   * call retainAll() to subset DB rows to those we wish to keep: the intersection of old and new.
+   * Then, call addAll() to add the diff(new - old) rows.
+   *
+   * <p>https://stackoverflow.com/questions/5587482/hibernate-a-collection-with-cascade-all-delete-orphan-was-no-longer-referenc
+   *
+   * @param emailDomains the new set of domains for this Institution
+   */
+  public DbInstitution setEmailDomains(final Set<DbInstitutionEmailDomain> emailDomains) {
+    final Set<DbInstitutionEmailDomain> attachedDomains =
+        emailDomains.stream()
+            .map(domain -> domain.setInstitution(this))
+            .collect(Collectors.toSet());
+    // modifies this set so that its value is the intersection of the two sets
```

Comment:
This looks like you're manually doing Hibernate's job. It seems really odd to me for an entity to take in a set of things it owns, then modify them and save them. I'd much rather see immutable email domain objects being pointed to by this set, unless you have a really urgent reason to work with domains but not the institution. Like where in the UI would you even have such an issue?